upcloud/service/managed_object_storage_test.go

```diff
@@ -558,30 +560,285 @@ func TestDeleteManagedObjectStorageUserAccessKey(t *testing.T) {
+			require.NoError(t, err)
+		}(storage.UUID)
+
+		policy, err := svc.CreateManagedObjectStoragePolicy(ctx, &request.CreateManagedObjectStoragePolicyRequest{
+			Name:        "testpolicy",
+			Description: "description2",
+			Document:    "%7B%22Version%22%3A%20%222012-10-17%22%2C%20%20%22Statement%22%3A%20%5B%7B%22Action%22%3A%20%5B%22iam%3AGetUser%22%5D%2C%20%22Resource%22%3A%20%22%2A%22%2C%20%22Effect%22%3A%20%22Allow%22%2C%20%22Sid%22%3A%20%22editor%22%7D%5D%7D",
+			ServiceUUID: storage.UUID,
+		})
+		require.NoError(t, err)
+		assert.Equal(t, policy.Name, "testpolicy")
+		assert.Equal(t, policy.Description, "description2")
+
+		err = svc.DeleteManagedObjectStoragePolicy(ctx, &request.DeleteManagedObjectStoragePolicyRequest{
+			ServiceUUID: storage.UUID,
+			Name:        policy.Name,
+		})
+		require.NoError(t, err)
+	})
+}
+
+func TestGetManagedObjectStoragePolicies(t *testing.T) {
+	record(t, "getmanagedobjectstoragepolicies", func(ctx context.Context, t *testing.T, rec *recorder.Recorder, svc *Service) {
+		storage, err := createManagedObjectStorage(ctx, svc)
+		require.NoError(t, err)
+
+		defer func(uuid string) {
+			err = deleteManagedObjectStorageAndUsers(ctx, svc, uuid)
+			require.NoError(t, err)
+		}(storage.UUID)
+
+		policy, err := svc.CreateManagedObjectStoragePolicy(ctx, &request.CreateManagedObjectStoragePolicyRequest{
+			Name:        "testpolicy",
+			Description: "description2",
+			Document:    "%7B%22Version%22%3A%20%222012-10-17%22%2C%20%20%22Statement%22%3A%20%5B%7B%22Action%22%3A%20%5B%22iam%3AGetUser%22%5D%2C%20%22Resource%22%3A%20%22%2A%22%2C%20%22Effect%22%3A%20%22Allow%22%2C%20%22Sid%22%3A%20%22editor%22%7D%5D%7D",
+			ServiceUUID: storage.UUID,
+		})
+		require.NoError(t, err)
+
+		policies, err := svc.GetManagedObjectStoragePolicies(ctx, &request.GetManagedObjectStoragePoliciesRequest{ServiceUUID: storage.UUID})
+		assert.NoError(t, err)
+		assert.Len(t, policies, 6)
+
+		err = svc.DeleteManagedObjectStoragePolicy(ctx, &request.DeleteManagedObjectStoragePolicyRequest{
+			ServiceUUID: storage.UUID,
+			Name:        policy.Name,
+		})
+		require.NoError(t, err)
+	})
+}
+
+func TestGetManagedObjectStoragePolicy(t *testing.T) {
+	record(t, "getmanagedobjectstoragepolicy", func(ctx context.Context, t *testing.T, rec *recorder.Recorder, svc *Service) {
+		storage, err := createManagedObjectStorage(ctx, svc)
+		require.NoError(t, err)
+
+		defer func(uuid string) {
+			err = deleteManagedObjectStorageAndUsers(ctx, svc, uuid)
+			require.NoError(t, err)
+		}(storage.UUID)
+
+		policy, err := svc.CreateManagedObjectStoragePolicy(ctx, &request.CreateManagedObjectStoragePolicyRequest{
+			Name:        "testpolicy",
+			ServiceUUID: storage.UUID,
+			Document:    "%7B%22Version%22%3A%20%222012-10-17%22%2C%20%20%22Statement%22%3A%20%5B%7B%22Action%22%3A%20%5B%22iam%3AGetUser%22%5D%2C%20%22Resource%22%3A%20%22%2A%22%2C%20%22Effect%22%3A%20%22Allow%22%2C%20%22Sid%22%3A%20%22editor%22%7D%5D%7D",
```

Comment:
Is it possible to pass JSON as input (instead of URL encoded JSON)? That would be more human readable 🤔

Reply:
I guess it has to be URL encoded in API requests and responses, but maybe we could automate the encode/decode part in the Go SDK 🤔

Reply:
Or probably less confusing if do not have automatic conversions 😅 In that case, we could use human readable JSON and encode functions in the tests and examples, but this is something we can also add later

Reply:
I kinda agree on the latest view that it could be a bit confusing in comparison to the API usage and documentation. End user should be able to just use `url.QueryEscape` & `url.QueryUnescape` from `net/url` while populating the request and reading the response. :)